packages/search/assets/js/containers/Search/Search.js

```diff
@@ -69,13 +65,20 @@ class Search extends React.Component<Props> {
                 name: routeName,
                 resultToRoute,
             },
-        } = this.indexes[result.document.index];
+        } = this.indexes[result.resourceKey];
+        const resultIdArray = result.id.split('::');
+        const resultId = resultIdArray[1] || null;
+        const resultLocale = resultIdArray[2] || null;
```

Comment:
Does the result not contain the resourceId, resourceKey and locale already? Else would change the API to return that.

```suggestion
        const resourceId = result.resourceId || null;
        const resourceId = result.resourceKey || null;
        const locale = result.locale || null;
```